In [1]:
import pandas as pd

In [2]:
#reading csv file
meta = pd.read_csv('../metadata.csv')

C:\Users\sohai\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3058: DtypeWarning: Columns (7,11,13,14,15,18,19,20,21,22,24,25,26,30,32) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


# Extracting Earthquake Metadata

Column Names

<ul>
    <li>'network_code', 'receiver_code', 'receiver_type', 'receiver_latitude',</li>
    <li>'receiver_longitude', 'receiver_elevation_m', 'p_arrival_sample',</li>
    <li>'p_status', 'p_weight', 'p_travel_sec', 's_arrival_sample', 's_status',</li>
    <li>'s_weight', 'source_id', 'source_origin_time',</li>
    <li>'source_origin_uncertainty_sec', 'source_latitude', 'source_longitude',</li>
    <li>'source_error_sec', 'source_gap_deg',</li>
    <li>'source_horizontal_uncertainty_km', 'source_depth_km',</li>
    <li>'source_depth_uncertainty_km', 'source_magnitude',</li>
    <li>'source_magnitude_type', 'source_magnitude_author',</li>
    <li>'source_mechanism_strike_dip_rake', 'source_distance_deg',</li>
    <li>'source_distance_km', 'back_azimuth_deg', 'snr_db', 'coda_end_sample',</li>
    <li>'trace_start_time', 'trace_category', 'trace_name'</li>
</ul>

In [3]:
#storing earthquake metadata to eq_meta dataframe
eq_meta = meta[meta['trace_category'] == 'earthquake_local']

In [4]:
eq_meta.dtypes

network_code                         object
receiver_code                        object
receiver_type                        object
receiver_latitude                   float64
receiver_longitude                  float64
receiver_elevation_m                float64
p_arrival_sample                    float64
p_status                             object
p_weight                            float64
p_travel_sec                        float64
s_arrival_sample                    float64
s_status                             object
s_weight                            float64
source_id                            object
source_origin_time                   object
source_origin_uncertainty_sec        object
source_latitude                     float64
source_longitude                    float64
source_error_sec                     object
source_gap_deg                       object
source_horizontal_uncertainty_km     object
source_depth_km                      object
source_depth_uncertainty_km     

In [5]:
#This function describe the magnitudes attribute
eq_meta['source_magnitude'].describe().apply(lambda x: format(x, 'f'))

count    1031908.000000
mean           1.528405
std            0.977819
min           -0.500000
25%            0.800000
50%            1.300000
75%            2.010000
max            7.900000
Name: source_magnitude, dtype: object

In [10]:
#nano
eq_meta[eq_meta['source_magnitude'] <1 ].shape

(347530, 35)

In [11]:
#micro
eq_meta[(eq_meta['source_magnitude']  >= 1) & (eq_meta['source_magnitude']  < 3)].shape

(579315, 35)

In [12]:
#small
eq_meta[(eq_meta['source_magnitude']  >= 3) & (eq_meta['source_magnitude']  < 5)].shape

(102997, 35)

In [13]:
#moderate
eq_meta[(eq_meta['source_magnitude']  >= 5) & (eq_meta['source_magnitude']  < 7)].shape

(2063, 35)

In [14]:
#major
eq_meta[(eq_meta['source_magnitude']  >= 7) & (eq_meta['source_magnitude']  < 9)].shape

(3, 35)

### Earthquake Magnitudes

<p>Nano earthquake when 1 < M</p>
<p>Micor earthquake when 1 <= M < 3 </p>
<p>Small earthquake when 3 <= M < 5 </p>
<p>moderate when 5 <= M < 7 </p>
<p>major when 7 <= M < 9 </p>
====================================
<p>Total Earthquake Data >= : 1,031,908  </p>
<p>Nano earthquake: 347,530 (33.67%)</p>
<p>Micro earthquake: 579,315 (56.13%)</p>
<p>Small earthquake: 102,997 (9.98%)</p>
<p>moderate: 2063 (0.19%)</p>
<p>major: 3</p>

In [10]:
sliced_eq_meta = pd.DataFrame(columns=eq_meta.columns)
sliced_eq_meta = sliced_eq_meta.append(eq_meta[(eq_meta['source_magnitude'] > 6) & (eq_meta['source_magnitude'] < 7)].iloc[0:4:2], ignore_index=True)

In [11]:
#This code creates sliced_eq_meta dataframe and appends to it 60 nano, 90 micro,  32 Small earthquakes, 15 moderate and 3 major

lower_limit = 0
#sliced_eq_meta = pd.DataFrame(columns=eq_meta.columns)

while(lower_limit <= 7):
    
    #for nano earthquakes
    if lower_limit == 0:
        sliced_eq_meta = sliced_eq_meta.append(eq_meta[eq_meta['source_magnitude']  < 1].iloc[0:120:2], ignore_index=True)
        lower_limit = lower_limit + 1
        
    #for micro earthquakes
    elif lower_limit == 1:
        sliced_eq_meta = sliced_eq_meta.append(eq_meta[(eq_meta['source_magnitude']  >= 1) & (eq_meta['source_magnitude']  < 3)].iloc[0:180:2], ignore_index=True)
        lower_limit = lower_limit + 2
    
    #for small magnitude earthquakes
    elif lower_limit == 3:
        sliced_eq_meta = sliced_eq_meta.append(eq_meta[(eq_meta['source_magnitude']  >= 3) & (eq_meta['source_magnitude']  < 5)].iloc[0:64:2], ignore_index=True)
        lower_limit = lower_limit + 2
        
    #for moderate magnitude earthquakes
    elif lower_limit== 5:
        sliced_eq_meta = sliced_eq_meta.append(eq_meta[(eq_meta['source_magnitude']  >= 5) & (eq_meta['source_magnitude']  < 7)].iloc[0:30:2], ignore_index=True)
        lower_limit = lower_limit + 2
        
    #for major earthquakes
    else:
        sliced_eq_meta = sliced_eq_meta.append(eq_meta[(eq_meta['source_magnitude']  >= 7) & (eq_meta['source_magnitude']  < 9)].iloc[:], ignore_index=True)
        lower_limit = lower_limit + 2

In [12]:
sliced_eq_meta[['source_magnitude']]

source_magnitude
0                6.30
1                6.20
2                0.87
3                0.70
4                0.90
..                ...
197              5.20
198              5.10
199              7.90
200              7.70
201              7.90

[202 rows x 1 columns]

In [14]:
sliced_eq_meta

network_code receiver_code receiver_type  receiver_latitude  \
0             PR          AGPR            BH          18.467493   
1             AV          AKLV            BH          54.161800   
2             TA          109C            BH          32.888900   
3             TA          A04A            BH          48.719700   
4             TA          A04A            BH          48.719700   
..           ...           ...           ...                ...   
197           LD          ACCN            BH          43.384300   
198           HT           AGG            HH          39.021100   
199           AK           FIB            BH          61.165600   
200            C          GO01            BH         -19.668500   
201           AK           SAW            BH          61.807000   

     receiver_longitude  receiver_elevation_m  p_arrival_sample p_status  \
0            -67.111183                 119.9             700.0   manual   
1           -165.957600                 551.0             700.0   manual   
2           -117.105100                 150.0             500.0   manual   
3           -122.707000                  23.0             800.0   manual   
4           -122.707000                  23.0             500.0   manual   
..                  ...                   ...               ...      ...   
197          -73.667800                 340.0             700.0   manual   
198           22.336000                 625.0             500.0   manual   
199         -150.177500                  62.0             600.0   manual   
200          -69.194200                3809.0             700.0   manual   
201         -148.331600                 781.0             400.0   manual   

     p_weight  p_travel_sec  ...  source_magnitude_author  \
0        0.50     13.420000  ...                     None   
1        0.50     32.730000  ...                     None   
2        0.64      3.760000  ...                       CI   
3        0.95      7.380000  ...                       UW   
4        0.95     10.680000  ...                       UW   
..        ...           ...  ...                      ...   
197      0.58     18.959999  ...                     None   
198      0.50     35.869999  ...                     None   
199      0.50     46.549999  ...                     None   
200      0.50     26.959999  ...                     None   
201      1.00     31.173000  ...                     None   

                      source_mechanism_strike_dip_rake  source_distance_deg  \
0    [[ 68.51999664  30.62000084  10.48999977]\n [3...               0.5700   
1    [[ 84.08000183  65.05999756 108.45999908]\n [2...               1.9100   
2                                                 None               0.1621   
3                                                 None               0.3771   
4                                                 None               0.4984   
..                                                 ...                  ...   
197                                               None               1.1280   
198                                               None               2.1500   
199                                               None               2.7400   
200  [[ 143.05000305    7.5        -116.65000153]\n...               1.5200   
201                                               None               1.7530   

    source_distance_km back_azimuth_deg  \
0                63.47           203.80   
1               211.92            26.90   
2                18.03           286.60   
3                41.90           233.20   
4                55.38           256.00   
..                 ...              ...   
197             125.36           358.00   
198             239.72            35.80   
199             304.83           206.90   
200             169.56            53.30   
201             195.52            13.72   

                                    snr_db  coda_end_sample  \
0    [85.80000305 82.599

In [13]:
#Saves sliced_eq_meta to a csv file
sliced_eq_meta.to_csv('metadata//eq_meta.csv', index=False)

# Extracting Noise Metadata

In [9]:
#storing noise metadata to noise_meta dataframe
noise_meta = meta[meta['trace_category'] == 'noise']

In [10]:
noise_meta.shape

(105885, 35)

Column Names

<ul>
    <li>'network_code', 'receiver_code', 'receiver_type', 'receiver_latitude',</li>
    <li>'receiver_longitude', 'receiver_elevation_m', 'p_arrival_sample',</li>
    <li>'p_status', 'p_weight', 'p_travel_sec', 's_arrival_sample', 's_status',</li>
    <li>'s_weight', 'source_id', 'source_origin_time',</li>
    <li>'source_origin_uncertainty_sec', 'source_latitude', 'source_longitude',</li>
    <li>'source_error_sec', 'source_gap_deg',</li>
    <li>'source_horizontal_uncertainty_km', 'source_depth_km',</li>
    <li>'source_depth_uncertainty_km', 'source_magnitude',</li>
    <li>'source_magnitude_type', 'source_magnitude_author',</li>
    <li>'source_mechanism_strike_dip_rake', 'source_distance_deg',</li>
    <li>'source_distance_km', 'back_azimuth_deg', 'snr_db', 'coda_end_sample',</li>
    <li>'trace_start_time', 'trace_category', 'trace_name'</li>
</ul>

In [30]:
noise_meta[['network_code', 'receiver_code', 'receiver_type', 'receiver_latitude',
'receiver_longitude','source_magnitude',
'source_magnitude_type','trace_start_time', 'trace_category', 'trace_name']].head()

network_code receiver_code receiver_type  receiver_latitude  \
1031908           TA          109C            HH            32.8889   
1031909           TA          109C            HH            32.8889   
1031910           TA          109C            HH            32.8889   
1031911           TA          109C            HH            32.8889   
1031912           TA          109C            HH            32.8889   

         receiver_longitude  source_magnitude source_magnitude_type  \
1031908           -117.1051               NaN                   NaN   
1031909           -117.1051               NaN                   NaN   
1031910           -117.1051               NaN                   NaN   
1031911           -117.1051               NaN                   NaN   
1031912           -117.1051               NaN                   NaN   

        trace_start_time trace_category               trace_name  
1031908              NaN          noise  109C.TA_201510210555_NO  
1031909              NaN          noise  109C.TA_201511061450_NO  
1031910              NaN          noise  109C.TA_201511070220_NO  
1031911              NaN          noise  109C.TA_201511140515_NO  
1031912              NaN          noise  109C.TA_201512251850_NO

In [11]:
sliced_noise_meta = pd.DataFrame(columns=eq_meta.columns)

#Selecting 200 noise samples with a jump of 8 samples
sliced_noise_meta = sliced_noise_meta.append(noise_meta.iloc[0:1600:8], ignore_index=True)

In [33]:
sliced_noise_meta

network_code receiver_code receiver_type  receiver_latitude  \
0             TA          109C            HH            32.8889   
1             TA          109C            HH            32.8889   
2             TA          109C            HH            32.8889   
3             TA          109C            HH            32.8889   
4             TA          109C            HH            32.8889   
..           ...           ...           ...                ...   
195           TA          214A            HH            31.9559   
196           TA          214A            HH            31.9559   
197           TA          214A            HH            31.9559   
198           TA          214A            HH            31.9559   
199           TA          214A            HH            31.9559   

     receiver_longitude  receiver_elevation_m  p_arrival_sample p_status  \
0             -117.1051                 150.0               NaN      NaN   
1             -117.1051                 150.0               NaN      NaN   
2             -117.1051                 150.0               NaN      NaN   
3             -117.1051                 150.0               NaN      NaN   
4             -117.1051                 150.0               NaN      NaN   
..                  ...                   ...               ...      ...   
195           -112.8115                 543.0               NaN      NaN   
196           -112.8115                 543.0               NaN      NaN   
197           -112.8115                 543.0               NaN      NaN   
198           -112.8115                 543.0               NaN      NaN   
199           -112.8115                 543.0               NaN      NaN   

     p_weight  p_travel_sec  ...  source_magnitude_author  \
0         NaN           NaN  ...                      NaN   
1         NaN           NaN  ...                      NaN   
2         NaN           NaN  ...                      NaN   
3         NaN           NaN  ...                      NaN   
4         NaN           NaN  ...                      NaN   
..        ...           ...  ...                      ...   
195       NaN           NaN  ...                      NaN   
196       NaN           NaN  ...                      NaN   
197       NaN           NaN  ...                      NaN   
198       NaN           NaN  ...                      NaN   
199       NaN           NaN  ...                      NaN   

    source_mechanism_strike_dip_rake  source_distance_deg source_distance_km  \
0                                NaN                  NaN                NaN   
1                                NaN                  NaN                NaN   
2                                NaN                  NaN                NaN   
3                                NaN                  NaN                NaN   
4                                NaN                  NaN                NaN   
..                               ...                  ...                ...   
195                              NaN                  NaN                NaN   
196                              NaN                  NaN                NaN   
197                              NaN                  NaN                NaN   
198                              NaN                  NaN                NaN   
199                              NaN                  NaN                NaN   

    back_azimuth_deg snr_db  coda_end_sample  trace_start_time trace_category  \
0                NaN    NaN              NaN               NaN          noise   
1                NaN    NaN              NaN               NaN          noise   
2                NaN    NaN              NaN               NaN          noise   
3                NaN    NaN              NaN               NaN          noise   
4                NaN    NaN              NaN               NaN          noise   
..               ...    ...              ...               ...            ...   
195              NaN    N

In [12]:
#Saves sliced_noise_meta to a csv file
sliced_noise_meta.to_csv('metadata//noise_meta.csv', index=False)